exp/runtime/hooks/api/v1alpha1/discovery_types.go

```diff
@@ -91,9 +91,14 @@ func Discovery(*DiscoveryRequest, *DiscoveryResponse) {}
 
 func init() {
 	catalogBuilder.RegisterHook(Discovery, &runtimecatalog.HookMeta{
-		Tags:        []string{"Discovery"},
-		Summary:     "Discovery hook",
-		Description: "Discovery hook discovers the supported hooks of a RuntimeExtension",
-		Singleton:   true,
+		Tags:    []string{"Discovery"},
+		Summary: "Cluster API Runtime will call this hook when a ExtensionConfig is reconciled",
```

Comment:
```suggestion
		Summary: "Cluster API Runtime will call this hook when an ExtensionConfig is reconciled.",
```
an ExtensionConfig + trailing dot (probably in other cases as well)

Reply:
I have removed all the trailing dot intentionally

Reply:
Just saw one somewhere